js_modules/dagit/packages/core/src/ticks/SensorDryRun.tsx

```diff
@@ -0,0 +1,361 @@
+                {error ||
+                (currentCursor ?? '') ===
+                  (sensorExecutionData?.evaluationResult?.cursor ?? 'testing1234') ? null : (
+                  <Box flex={{direction: 'row', gap: 8, alignItems: 'center'}}>
+                    <Button
+                      disabled={['Persisting', 'Persisted'].includes(cursorState)}
+                      loading={cursorState === 'Persisting'}
+                      onClick={onPersistCursorValue}
+                    >
+                      <span data-testid={testId('persist-cursor')}>
+                        {cursorState === 'Persisting'
+                          ? 'Persisting'
+                          : cursorState === 'Persisted'
+                          ? 'Persisted'
+                          : 'Persist computed cursor value'}
+                      </span>
+                    </Button>
+                    {cursorState === 'Persisted' ? (
+                      <Icon name="check_circle" color={Colors.Green500} />
+                    ) : null}
+                  </Box>
+                )}
+              </div>
+            </Grid>
+            {error ? (
+              <div>
+                <PythonErrorInfo error={error} />
+              </div>
+            ) : null}
+            {didSkip ? (
+              <div>
+                <Subheading>Skip Reason</Subheading>
+                <div>{sensorExecutionData?.evaluationResult?.skipReason}</div>
+              </div>
+            ) : null}
+            {numRunRequests && runRequests ? (
+              <RunRequestTable
+                runRequests={runRequests}
+                name={name}
+                jobName={jobName}
+                isJob={true}
+                repoAddress={repoAddress}
+              />
+            ) : null}
+          </Box>
+        </Box>
+      );
+    }
+    if (submitting) {
+      return (
+        <Box flex={{direction: 'row', gap: 8, justifyContent: 'center', alignItems: 'center'}}>
+          <Spinner purpose="body-text" />
+          <div>Evaluating sensor</div>
+        </Box>
+      );
+    } else {
+      return (
+        <Box flex={{direction: 'column', gap: 8}}>
+          <div>Cursor</div>
+          <TextInput
+            value={cursor}
+            onChange={(e) => setCursor(e.target.value)}
+            data-testid={testId('cursor-input')}
+          />
+        </Box>
+      );
+    }
+  }, [
+    sensorExecutionData,
+    error,
+    submitting,
+    currentCursor,
+    cursorState,
+    onPersistCursorValue,
+    name,
+    jobName,
+    repoAddress,
+    cursor,
+  ]);
+
+  return (
+    <Dialog
+      isOpen={isOpen}
+      onClose={onClose}
+      style={{width: '70vw', display: 'flex'}}
+      title={
+        <Box flex={{direction: 'row', gap: 8, alignItems: 'center'}}>
+          <Icon name="sensors" />
+          <span>{name}</span>
+        </Box>
+      }
+    >
+      <DialogBody>
+        <div style={{minHeight: '300px'}}>{content}</div>
```

Comment:
You might consider moving the `content` part into its own component, that way you don't have to perform all of the state setup above unless the Dialog is actually opened. (The contents shouldn't render until the Dialog is open.)